scripts/generate_rust_analyzer.py

```diff
@@ -65,7 +66,7 @@ def append_crate(display_name, root_module, deps, cfg=[], is_workspace_member=Tr
         [],
         is_proc_macro=True,
     )
-    crates[-1]["proc_macro_dylib_path"] = "rust/libmacros.so"
+    crates[-1]["proc_macro_dylib_path"] = f"{objtree}/rust/libmacros.so"
```

Comment:
Maybe use `objtree / "rust" / "libmacros.so"` instead? Otherwise an empty objtree value would result in the absolute path `/rust/libmacros.so`, which is incorrect. I don't know if an empty objtree value is possible though.

Reply:
`objtree` is a required param in the script. For an in source build the Makefile sets the value to `.` and if its an out of source the value is the same as the path specified in the `O=` parameter. So this should never be empty?

Reply:
Alright, in that case the current code is fine.